core/src/main/java/org/apache/iceberg/InheritableMetadataFactory.java

```diff
@@ -75,6 +75,13 @@ public <F extends ContentFile<F>> ManifestEntry<F> apply(ManifestEntry<F> manife
         manifestEntry.setSequenceNumber(sequenceNumber);
       }
 
+      // in v1 tables, the file sequence number is not persisted and can be safely defaulted to 0
+      // in v2 tables, the file sequence number should be inherited iff the entry status is ADDED
```

Comment:
typo: iff -> if?

Reply:
That was actually deliberate to indicate that's the only case when it is safe (iff means if and only if).

Reply:
Thanks for the explanation!